base/server/python/pki/server/pkidestroy.py

```diff
@@ -177,11 +199,13 @@ def main(argv):
         config.PKI_DEPLOYMENT_DEFAULT_CONFIGURATION_FILE
 
     # establish complete path to previously deployed configuration file
-    config.user_deployment_cfg =\
-        deployed_pki_subsystem_path + "/" +\
-        "registry" + "/" +\
-        deployer.subsystem_name.lower() + "/" +\
-        config.USER_DEPLOYMENT_CONFIGURATION
+    if not force_destroy:
```

Comment:
Could you clarify what this code means here?

Reply:
The lines inside `if` tries to load the user deployment config file. During normal destroy, they read the file and load corresponding entries to master dictionary. `parser.validate()` tries to validate whether this cfg file exists and throws error. NOTE: this method is used by `pkispawn` too.

When forced destroy, we don't care for this config file.

However, I can modify a little and do this:
````python
config.user_deployment_cfg = os.path.join(
            deployed_pki_subsystem_path,
            "registry",
            deployer.subsystem_name.lower(),
            config.USER_DEPLOYMENT_CONFIGURATION
        )

if force_destroy and not os.path.exists(config.user_deployment_cfg):
    # During force destroy, try to load the file. If file doesn't exist, we ignore it
    config.user_deployment_cfg = None
````